### DISCUSSION TITLE
Code Listing

### DISCUSSION THREAD

**Discussion by lu-kas** [executablebooks/meta#223] [Q&A] [answered]
Dear all,

first, thanks to the developers for this great tool! 

I would like to include the listing of a file (e.g. a source code file) into the jupyter notebook. Like the listing environment in LaTeX. Is this somehow possible?  

NB, I know that I could copy the content of the file into the notebook, but this way I would have to make sure that it is in sync with the actual file.

Thanks!

**Comment by jstac** [comment]
@mmcky is working on this issue but I'm not sure what the status is.

**Comment by mmcky** [comment, marked as answer]
@lu-kas does the contents of that file need to be executable? 

If **not** you can use the [sphinx `literalinclude` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude) to bring in the contents from a file (but this can't be used to bring in `executable code` just contents of files for viewing that may or may not need syntax highlighting)

````
```{literalinclude} <path-to-file>
```
````

We are working on support in `jupyter-book` to bring in code from a file through a `:file:` option to `code-cell` directives. It is currently in `PR` phase and not yet merged -- but it is coming. You can follow progress: https://github.com/executablebooks/MyST-NB/pull/286

**Reply by lu-kas** [reply]
Thanks!

This is what I was looking for -- just a passive (non executed) listing of a file. I have to admit, that I should have had a closer look into the documentation. I have not extended my search scope to the tools jupyterbook builds upon. 

It is even possible to select lines, add line numbers and reference to that listing. Great!

As this capability is not part of the jupyterbook documentation, I could offer to extend the docs with a short description. Do you think, this is valuable also for other users?

**Reply by choldgraf** [reply]
@lu-kas go for it!
